### ISSUE TITLE
During development, Flutter app on iOS occasionally won't launch

### ISSUE THREAD

**Issue by lukepighetti** [flutter/flutter#34959]
I just added assets to my project and now the app won't launch on iOS. I haven't tested on Android. If I remove the assets from `pubspec.yaml` it will launch.

I added a 7kb png file like this (green).

<img width="407" alt="Screen Shot 2019-06-23 at 3 19 36 PM" src="https://user-images.githubusercontent.com/666539/59980904-5136f780-95ca-11e9-97cd-de1737b428c3.png">

```
flutter: 
  uses-material-design: true 
  assets:
    - assets/send.png

```

Error message: 
```
Launching lib/main.dart on Luke's iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: TA8U33SGB8
Xcode build done.                                            8.5s
Configuring the default Firebase app...
6.3.0 - [Firebase/Core][I-COR000004] App with name __FIRAPP_DEFAULT does not exist.
6.3.0 - [Firebase/Analytics][I-ACS023007] Analytics v.60002000 started
6.3.0 - [Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
Configured the default Firebase app __FIRAPP_DEFAULT.
Could not update files on device: OS Error: Connection reset by peer, errno = 54
Exited (sigterm)
```

Deps
```
dependencies:
  flutter:
    sdk: flutter

  crypto: ^2.0.6
  cupertino_icons: ^0.1.2
  firebase_auth: ^0.11.1+7
  firebase_database: ^3.0.3
  hydrated: ^1.2.3
  intl: ^0.15.8
  provider: ^3.0.0+1
  rxdart: ^0.22.0
  timeago: ^2.0.16
```

Doctor
```
[✓] Flutter (Channel dev, v1.7.8, on Mac OS X 10.14.5 18F132, locale en-US)
    • Flutter version 1.7.8 at /Users/lukepighetti/flutter
    • Framework revision d51fd86cdb (2 days ago), 2019-06-21 22:31:55 -0400
    • Engine revision d004bcd4d6
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/lukepighetti/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/lukepighetti/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2.1, Build version 10E1001
    • CocoaPods version 1.7.2

[✓] iOS tools - develop for iOS devices
    • ios-deploy 1.9.4

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 26.0.1
    • Dart plugin version 173.4700
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1024-b01)

[✓] VS Code (version 1.35.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.1.0

[✓] Connected device (3 available)
    • Luke's iPhone • 0bc0c4864d8e08aac4456893cded68bf0b6767d5 • ios            • iOS 12.3.2
    • macOS         • macOS                                    • darwin-x64     • Mac OS X 10.14.5 18F132
    • web           • web                                      • web-javascript • Google Chrome 75.0.3770.100 


• No issues found!
```

**Comment by lukepighetti** [comment]
Alright, I did a restart and it worked. But then after a little bit it quit working again.

**Comment by jonahwilliams** [comment]
Is it the action of adding this asset and hot reloading/restarting, or only the presence of it? Does this happen with other assets or in a fresh flutter create project?

**Comment by coastalgit** [comment]
I am experiencing the exact same error when attempting a launch from VSCode to a device (but works OK with sim), although with no asset adaption involved. The error only started occurring when I upgraded the device to iOS 12.3.1 When running the same code to the device **directly from Xcode** it works OK.
```
Launching lib/main.dart on BF iPhone 6 in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: 59J6XF5LU4
Xcode build done.                                           12.4s
Configuring the default Firebase app...
6.3.0 - [Firebase/Core][I-COR000004] App with name __FIRAPP_DEFAULT does not exist.
Configured the default Firebase app __FIRAPP_DEFAULT.
6.3.0 - [Firebase/Messaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
6.3.0 - [Firebase/Analytics][I-ACS023007] Analytics v.60002000 started
6.3.0 - [Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
Could not update files on device: OS Error: Connection reset by peer, errno = 54
Exited (sigterm)
```
Reboot of device and MacBook has not resolved either.

**Comment by lukepighetti** [comment]
I don't have anything particularly helpful to add to this but I've been experiencing this on macOS since I started using Flutter in mid 2018. I see it come up in your average project that has a few screens and a few deps and possibly some assets. I haven't been able to figure out what causes it. The fact that you can spam `flutter run` and it works 30% of the time makes it really hard to track down since it's an intermittent problem.

**Comment by lukepighetti** [comment]
@jonahwilliams I'm sorry I don't have any conclusive answer to that question.

**Comment by kangwang1988** [comment]
@lukepighetti 
Can you do some experiment to check whether it will work fine if you change to another picture, for example, a smaller png or jpg? It seems related to the send.png.

**Comment by lukepighetti** [comment]
Quick update. I just had a new project that was working fine until I added nine ~120kb *.png files to `assets/images`, and added this line to `pubspec.yaml`. Before this I had multiple Font assets that were working perfectly in `fonts/`

```
flutter:
  uses-material-design: false
  assets:
    - assets/images/
```

It `Exited (sigterm)` with no error three times, launched once. Then I had to restart because I had typed in the asset path incorrectly within my Flutter app. It then `Exited (sigterm)` once with no error. Then it gave a `54` error four times stating that it couldn't upload files to device (lost the exact error message, sorry). And now it says "Could not install XYZ. Try launching Xcode and selecting "Product > Run"".

Open XCode, click Play. `Build Failed`. Code signing error, need a team. (This didn't come up before?) Selected my team, click Play. Builds perfectly and quickly. Quit XCode, go back to VSC, hit debug. Starts no problem.

Kill from VSC and debug again, `Exited (Sigterm)` with no error.
Run from VSC again, `Exited (Sigterm)` with no error.
Run from VSC again, `Exited (Sigterm)` with error `Could not update files on device: OS Error: Connection reset by peer, errno = 54`.
Run from VSC again, starts.

Frustrating!

**Comment by jonahwilliams** [comment]
I wonder if we're running out of space on the device. @bkonyi did we get that change to the devfs landed that reports filesystem exceptions?

**Comment by bkonyi** [comment]
Yup, should have landed in https://github.com/dart-lang/sdk/commit/34a6218f8e5394eb8182381b84c480c01f6878d3

**Comment by lukepighetti** [comment]
Fwiw I'm on `beta` channel in that previous comment

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.7.8+hotfix.2, on Mac OS X 10.14.5 18F132, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
[✓] iOS tools - develop for iOS devices
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.36.0)
[✓] Connected device (1 available)

• No issues found!
```

**Comment by lukepighetti** [comment]
If there's anything I can do to assist please don't hesitate to ask.

**Comment by lukepighetti** [comment]
Cross post just in case it's helpful: https://www.reddit.com/r/FlutterDev/comments/cbow7f/anyone_else_having_problems_with_ios_attaching/

**Comment by lukepighetti** [comment]
FWIW I just tested this with the iPhone plugged into a USB hub and plugged directly into my iMac and the results are identical.

**Comment by lukepighetti** [comment]
@jonahwilliams You talking normal "disk" space? I have 4 gb free on my device.

**Comment by lukepighetti** [comment]
Hi again. I just ran into a situation where I refactored a project and I've tried to attach 20 times and each time it fails with error 54. Requesting more attention/suggestions for ways I can help fill out this issue with more information. @jonahwilliams @bkonyi

**Comment by lukepighetti** [comment]
@coastalgit did you find a solution?

**Comment by lukepighetti** [comment]
Still having this issue on master branch

```
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, v1.8.2-pre.30, on Mac OS X 10.14.5 18F132, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.2.1)
[✓] iOS tools - develop for iOS devices
[✓] Chrome - develop for the web
[✓] Android Studio (version 3.1)
[✓] VS Code (version 1.36.1)
[✓] Connected device (3 available)

• No issues found!
```

**Comment by coastalgit** [comment]
I'm afraid not. Alas, I've been working and testing predominantly on
Android for the moment.

**Comment by lukepighetti** [comment]
Getting this error when doing `flutter run --profile`. Not sure if it's connected.

```
Failed to find snapshot: /var/containers/Bundle/Application/CF6FF25D-DE07-48EF-9D22-9398596818BC/Runner.app/Frameworks/App.framework/flutter_assets/kernel_blob.bin
[VERBOSE-2:engine.cc(118)] Engine run configuration was invalid.        
[VERBOSE-2:FlutterEngine.mm(323)] Could not launch engine with configuration.
Installing and launching...                                         9.8s

```

**Comment by lukepighetti** [comment]
@jonahwilliams @bkonyi @kangwang1988 friendly ping

**Comment by lukepighetti** [comment]
Alright, I'm sorry for doing this, but this has gone on long enough. @abarth @Hixie @HansMuller @jonahwilliams @cbracken @devoncarew @goderbauer @tvolkert @jason-simmons @xster

**Comment by Hixie** [comment]
@lukepighetti We really have no idea what could be causing this. We haven't been able to reproduce it. I recommend trying to debug the issue (e.g. try different devices, different hosts, see if there's specific steps to reproduce from a clean `flutter create`, try enabling verbose mode and seeing what's failing, see if you can reproduce without the `flutter` tool, etc).

**Comment by lukepighetti** [comment]
Thanks for taking the time to respond @Hixie, I will close unless I come up with a repro.

**Comment by thavlik** [comment]
I experience this daily as well. The only way I can launch my flutter app is through XCode. Did you come up with the repro?

**Comment by lukepighetti** [comment]
I have been unable to come up with a repro because the issue is always present.

**Comment by tvolkert** [comment]
@lukepighetti does the unmodified `flutter create` app work for you on iOS (it includes assets)?

**Comment by lukepighetti** [comment]
It's sporadic. Sometimes it works flawlessly, sometimes it doesn't attach.

**Comment by tvolkert** [comment]
Reading the history, it sounds like this is what we know - can you confirm that:

1. You see this failure when attempting to `flutter run` from the CLI
1. You only see it on a iOS physical device (simulators work ok) and haven't tested on Android
1. This problem occurs for you frequently (~80-90% of the time)
1. It occurs in projects with and without assets, so assets seem to be unrelated to the issue
1. The logs when this issue happens say "Could not update files on device: OS Error: Connection reset by peer, errno = 54", then "Exited (sigterm)"
 
Is that all correct?  I want to make sure that the history is all still accurate, and we're not conflating 2 or more problems herein.

**Comment by lukepighetti** [comment]
1. It happens when I run debug in VSC or from the CLI. If I launch from XCode it works every time.
2. I have not tested simulators exhaustively.
3. It's sporadic. It occurs 90% of the time 20% of the time. When it happens, it happens so regularly that it can take an hour to get a single attached debug instance. When it's not happening, there's no issue.
4. That appears to be correct. Not related to assets.
5. That is the exact error I get 80% of the time it fails. Sometimes it fails silently with no error message.

**Comment by tvolkert** [comment]
Thanks!  And following up on https://github.com/flutter/flutter/issues/34959#issuecomment-511138855, can you run successfully using `flutter run --release`?

**Comment by lukepighetti** [comment]
I believe I can say with confidence that `flutter run --release` does not improve the situation. Due to the sporadic nature of this bug it's hard to test but I would say i'm 65% certain that is accurate.

**Comment by tvolkert** [comment]
Hmm.  The fact that it runs from Xcode (which just builds and runs the app without trying to connect to the Dart VM service protocol) implies that the problem may be in the VM service connection code path (it sounds like there's some race condition).  Given that `flutter run --release` doesn't attempt to connect to the VM service (the VM service isn't running at all in release mode), I'd expect release mode to work ok.

Unfortunately, release mode builds also take longer, so attempting to reproduce this by spamming `flutter run --release` will require more patience on your part...

Would you be willing to see for sure if you can reproduce these problems in release mode?

**Comment by lukepighetti** [comment]
I have to backtrack, I was able to test today and it appears that `flutter run --release` does in fact work as expected. Said another way, it does not appear to exhibit any issues.

**Comment by lukepighetti** [comment]
It definitely feels like a race condition.

**Comment by tvolkert** [comment]
@lukepighetti out of curiosity, which iPhone version are you testing against?  Is it 32 or 64 bit?

**Comment by aam** [comment]
Are there any crashes reported in Console for the ios device when you encounter `sigterm` exit?
Can you share logs(post a link to [gist](https://gist.github.com/)) of running with `--verbose-system-logs` option(f.e. `$FH/flutter/bin/flutter run -v --verbose-system-logs`)

**Comment by lukepighetti** [comment]
iPhone 8+ on iOS 12.3.2

logs: https://gist.github.com/lukepighetti/981ac721679294e8653ef2ddd8aec393

**Comment by lukepighetti** [comment]
Just pulled down a project and I've been sitting here trying to launch this thing for the last 30 minutes. I'll gift someone a month of Discord Nitro if they can help me nail down a solution.

**Comment by aam** [comment]
> Are there any crashes reported in Console for the ios device when you encounter sigterm exit?
I mean the Console app in the Utilities folder on mac
<img width="954" alt="Screen Shot 2019-09-10 at 7 51 02 AM" src="https://user-images.githubusercontent.com/381137/64624721-eebaef80-d39f-11e9-938a-6a901c5e1175.png">

It looks like the app on the device crashes which is why vmservice silently disconnects, hopefully the logs will give us some clues as to why.

**Comment by lukepighetti** [comment]
I'm not sure what I'm looking for but I'm happy to get on audio/video chat to go through the motions.

**Comment by lukepighetti** [comment]
Top is what I get when it succeeds

<img width="899" alt="Screen Shot 2019-09-10 at 1 23 23 PM" src="https://user-images.githubusercontent.com/666539/64635792-6a339580-d3ce-11e9-8065-97ecad1c640a.png">

Bottom is what I get when it fails
<img width="881" alt="Screen Shot 2019-09-10 at 1 24 27 PM" src="https://user-images.githubusercontent.com/666539/64635808-71f33a00-d3ce-11e9-8b84-4f7126183a10.png">


Honestly a 15 min group chat would probably save us both hours of time. I'm on the Flutter Discord.

**Comment by aam** [comment]
I think I just had it happen to me as well - the application on the device(flutter sample project on iphone xr 12.4.1) was not affected - it didn't crash and was launched without problems. The failure seems to be during copying the assets over to the connected device:
```
$ $FH/flutter/bin/flutter run -v --verbose-system-logs 
...
[        ] Updating files
[   +6 ms] Error writing
"build/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf" to
DevFS: OS Error: Connection reset by peer, errno = 54
[        ] Could not update files on device: OS Error: Connection reset by peer,
errno = 54
[   +1 ms] Syncing files to device yellow XR... (completed in 1,400ms)
[   +1 ms] "flutter run" took 33,655ms.

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      RunCommand.runCommand
(package:flutter_tools/src/commands/run.dart:465:7)
```

I can't reproduce the failure again though.

**Comment by lukepighetti** [comment]
@aam Reproducing it is almost impossible, it will just happen in random streaks.

**Comment by miguelpruivo** [comment]
I'm on same boat @lukepighetti, this looks like a _no reason_ issue that happens without a purpose at all. I can't even know what makes it happen, but when it does, it's there for good and will alternate between `connection reset by peer`, `exit sigterm` or even no info at all.

**Comment by thavlik** [comment]
None of them work for me, either. I am getting all of these errors in sequence with every attempt. Check https://github.com/flutter/flutter/issues/38677#issuecomment-528980904

I encounter this problem on multiple independent hardware setups (different Macs, iPads, cables, adapters, everything). It has persisted across major versions of flutter.

**Comment by aam** [comment]
I think we should be able to add more logging around https://github.com/dart-lang/sdk/blob/master/runtime/bin/vmservice/server.dart#L263 to see why vmservice http connection is getting closed during those post requests.

If http-posting large files on ios is so flaky we could add retry process or we could send them in chunks, concatenating chunks together in vmservice.

**Comment by tvolkert** [comment]
@aam perhaps you could create a patch that adds _super_ verbose logging and have people here locally patch it in and report back with their logs?

**Comment by aam** [comment]
@tvolkert wrote:
> @aam perhaps you could create a patch that adds _super_ verbose logging and have people here locally patch it in and report back with their logs?

Perhaps, but http server that needs instrumentation is part of flutter engine(rather than flutter dart framework code), so applying such a patch would require some effort(rebuilding flutter engine).

**Comment by tvolkert** [comment]
Ah, that does make it tricker.  Easiest way to do it (which isn't all that easy) would be to commit a patch on a branch in the engine repo, then roll to that engine on a branch on the framework repo, then have people clone your branch and try it out.

**Comment by lukepighetti** [comment]
Happy to help as long as someone can give me high level overview of patch steps

**Comment by thavlik** [comment]
Same!

**Comment by mwawrusch** [comment]
Happens to me too. both flutter 1.7 and 1.9.

**Comment by lukepighetti** [comment]
I've given up on developing on device for the time being. I'm using Simulator now until we can get more visibility on this show-stopping bug.

**Comment by tvolkert** [comment]
@lukepighetti I just tagged this with labels to give it more visibility to the team

**Comment by lukepighetti** [comment]
Just wanted to make sure this comment on the other issue got some visibility: https://github.com/flutter/flutter/issues/35989#issuecomment-531786002

**Comment by tvolkert** [comment]
@lukepighetti are you saying that the following fixes this completely for you?

```sh
$ export NO_PROXY=localhost,127.0.0.1 && flutter run
```

**Comment by aam** [comment]
I have added temporary channel(branch) `verbose-put` which uses engine(dart/vm) with more verbose logging around handling of dart vmserivce PUT request that seems to be causing issues. If folks who experience the problem could share the logs of `flutter run -v` (specifically lines prefixed with `vmservice PUT request` between `Updating files` and `DevFS: Sync finished`) via gist, that would help.

You switch to this channel by fetching latest flutter repo (https://github.com/flutter/flutter) and either doing `git checkout verbose-put` or `bin/flutter channel verbose-put`.

To clarify - this is not a fix to the problem, this is just to hopefully understand why and where this failure happen.

**Comment by lukepighetti** [comment]
@tvolkert @aam when the error comes up again I will fiddle around, thank you!

**Comment by lukepighetti** [comment]
Uuuuuuuuhhhhhhhhh

**Comment by tvolkert** [comment]
That's an over-aggressive bot 🙂

**Comment by alan-criton** [comment]
I've been struggling with this similar error on 1.7.2+hotfix.4 and 1.9.1+hotfix.2 when trying to build in development mode on iphone 7 and iphone X (both running ios 12.x). 

Gist from verbose-put: https://gist.github.com/alan-criton/d4138de4d7e60e9682b9e41088cd496c

**Comment by aam** [comment]
Thank you for the logs @alan-criton .
From the logs it looks like server didn't experience any problems up to the point when connection was dropped. Only theory I have for that is that proxy or http connection intermediary that connects host and device can do that and flutter tool needs to work around that by retrying or breaking large files into chunks.

**Comment by aam** [comment]
On further inspection of the logs it seems that the culprit is the fact that we have up to [6 concurrent writes](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/devfs.dart#L273)(http put) happening at the same time. It sounds like this might be too much in some cases - reducing to 1 fixed the problem that @alan-criton was having.
Number 6 was introduced in https://github.com/flutter/flutter/pull/5287, not sure how it was derived.

**Comment by lukepighetti** [comment]
@aam is there any way we can reduce that number and do a hotfix on stable branch? I'd rather have uploads that take 6x longer than ones that succeed 10% of the time when it's problematic.

**Comment by lukepighetti** [comment]
https://gist.github.com/lukepighetti/1716ff955f9defc4480c9da746190029

**Comment by lukepighetti** [comment]
Some more logs from trying to debug/attach in VSCode

```
Launching lib/main.dart on Luke's iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: TA8U33SGB8
Xcode build done.                                           19.0s
Configuring the default Firebase app...
Configured the default Firebase app __FIRAPP_DEFAULT.
timeout, returning true
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/authentication/google.png
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/authentication/fake.png
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/test/video.mp4
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
Exited (sigterm)
```

```
Launching lib/main.dart on Luke's iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: TA8U33SGB8
Xcode build done.                                           19.4s
Configuring the default Firebase app...
Configured the default Firebase app __FIRAPP_DEFAULT.
timeout, returning true
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/authentication/google.png
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/authentication/fake.png
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/assets/test/video.mp4
flutter: vmservice PUT request: fsName: skatex, fsPath: null, fsUri: build/flutter_assets/packages/cupertino_icons/assets/CupertinoIcons.ttf
{"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
done
flutter: vmservice PUT request: result {"jsonrpc":"2.0","id":"","result":{"type":"Success"}}
Could not update files on device: OS Error: Connection reset by peer, errno = 54
Exited (sigterm)
```

**Comment by aam** [comment]
> Can I just edit kMaxInFlight in my local flutter repo or do I have to do some kind of build step? Unfortunately I cannot develop on Simulator because my app uses textures.

Sure, you can do that, no build step is needed, but you need to remove cached version of flutter_tools: `$ rm $FH/flutter/bin/cache/flutter_tools`

I am hesitant to reduce `kMaxInFlight` because it seems that it will mask real culprit. I am hopefully we find that culprit shortly.

**Comment by lukepighetti** [comment]
```
cd ~/flutter
git checkout -b max-in-flight
code packages/flutter_tools/lib/src/devfs.dart # edit kMaxInFlight from 6 to 1
rm bin/cache/flutter_tools*
```

This seems to have resolved the issue. As a note, I tried setting it to `4` and it failed. Oh my goodness seeing this thing attach reliably is unbelievable.

**Comment by aam** [comment]
For those who are experiencing this problem, I updated `verbose_put` channel/branch so it has retry logic with 500ms sleep between retries. I am curious to hear whether it helps.

**Comment by dev-vinicius** [comment]
I add the file "GoogleService-Info.plist" with VSCode and get the same error. When I open the project in Xcode (ios folder) the file was not there for some reason. So I add the same file with Xcode and when I run the project on VSCode, it worked.

**Comment by ElectrifyPowr** [comment]
I had this very weird issue where my app on iOS wouldn't build but the app had no problems on Android.
I tried a lot of different things but none of them worked, including recreating a new project from scratch. I finally found the error when I tested out different assets.
All assets were actually working except for one image. When I removed the image from the assets the iOS project built successfully, but if I included it the app would not build.

To be more specific, the image that caused the problem was image taken by a camera of an Android phone. All the other assets were taken from somewhere else.
Now, I do not know if Android has some special image encoding that Xcode _doesn't like_ but this issue was really frustrating because the error message did not indicate anything about the real issue.
The error was simply **Command PhaseScriptExecution failed with a nonzero exit code** without further explanation...

So, what I did in the end, a _simple hack_ was open the image on a Mac, take a screenshot of the image with the screenshot tool of the Mac and then simply use that image instead. This fixed the issue and the iOS app build successfully.
